table/tables/tables.go

```diff
@@ -1091,8 +1091,9 @@ func (t *TableCommon) Allocators(ctx sessionctx.Context) autoid.Allocators {
 }
 
 // RebaseAutoID implements table.Table RebaseAutoID interface.
-func (t *TableCommon) RebaseAutoID(ctx sessionctx.Context, newBase int64, isSetStep bool) error {
-	return t.Allocators(ctx).Get(autoid.RowIDAllocType).Rebase(t.tableID, newBase, isSetStep)
+// Both auto-increment and auto-random can use this function to do rebase on explicit newBase value (without shadow bits).
+func (t *TableCommon) RebaseAutoID(ctx sessionctx.Context, newBase int64, isSetStep bool, tp autoid.AllocatorType) error {
+	return t.Allocators(ctx).Get(tp).Rebase(t.tableID, newBase, isSetStep)
```

Comment:
We can't rebase the ID of AutoRandom in this way, since the allocator should be rebased only on 'incremental part' of AutoRandom values, please check the code here:

https://github.com/pingcap/tidb/blob/35a801bc9ba21a8013abaa2c9143bab7f083ebc5/executor/insert_common.go#L906

Reply:
It's a different logic process.

I think the auto_random Num is explicit without mask